### DISCUSSION TITLE
Error when opening Child modal

### DISCUSSION THREAD

**Discussion by bdfejes** [wire-elements/modal#362] [Q&A]
I have migrated my app from v1 to v2 and livewire v2 to v3 (Laravel 10, PHP 8.2) and now i receive an error every time i open a child modal. The parent modal disappears, but the child doesn't show up. In the console i got an error:

livewire.js?id=f41737f6:6605 Uncaught TypeError: Cannot read properties of null (reading 'before')
    at Block.appendChild (livewire.js?id=f41737f6:6605:23)
    at patchChildren (livewire.js?id=f41737f6:6433:21)
    at patchChildren (livewire.js?id=f41737f6:6475:11)
    at patch (livewire.js?id=f41737f6:6371:7)
    at patchChildren (livewire.js?id=f41737f6:6522:9)
    at patch (livewire.js?id=f41737f6:6371:7)
    at patchChildren (livewire.js?id=f41737f6:6522:9)
    at patch (livewire.js?id=f41737f6:6371:7)
    at patchChildren (livewire.js?id=f41737f6:6522:9)
    at patch (livewire.js?id=f41737f6:6371:7)

I can't figure it out, what can cause this.
Also if i close a modal from a component using the closeModalWithEvent, and  try to open another modal i got stuck at Livewire Loading state without any error, the modal opens but my loading screen doesn't disappear. I don't know this is related to the first error or not.

**Comment by PhiloNL** [comment]
Would you mind testing the modal with a clean installation and verifying if the problem still occurs? Often problems are caused by project-specific cases that are difficult to debug without granting access to the code of the project.

**Reply by bdfejes** [reply]
I will do it when I have the time for it. I understand there can be a lot of project-specific factors. I have to add that I modified the modal.blade.php file to use Bootstrap 5, but I already tried it with the original Tailwind classes, and the error still occurs. Anyway, those classes should only affect the modal visually, not functionally, shouldn't they?

**Reply by camerontucker** [reply]
I'm also hitting this after updating to livewire v3, but only in production... 

Is this relevant? https://github.com/livewire/livewire/discussions/6663

**Comment by yabdab** [comment]
I am getting same error, was this ever resolved?

In my case I have buttons to open multiple child modals inside the parent modal. If I closeModal on a child , and then try to open a different child, then I get same error as above and just the overlay.

**Reply by yabdab** [reply]
this is a fresh install using livewire 3 and laravel 11

**Reply by camerontucker** [reply]
I believe I resolved with config/livewire.php

`'inject_morph_markers' => false`

**Reply by yabdab** [reply]
It looks like one of my modal views had a broken DOM (missing closing tag). Everything works good once that was fixed.

**Reply by insulae** [reply]
> 'inject_morph_markers' => false

I had the same problem and solved it using this line. I don't think it's the best option, since it's deactivating something important (I think). It would be great to find another solution. In the meantime, I'll test how the entire system behaves with false. Thanks!
